addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar.scss

```diff
@@ -34,19 +34,21 @@
     display: flex;
     flex: 1;
     justify-content: flex-end;
+
+    &.o-has-flap {
+        // at least 40px to prevent overlapping with 'close' button (see .o_documents_close_chatter)
+        margin-right: 60px;
+    }
```

Comment:
Let's now move xml of flap in the chatter topbar, so that you no longer need hard-coded margin (and you can rely on default `position: static` with same `z-index` behaviour of HTML documents that prevents text and sibling nodes from overlapping).